madara-prover-rpc-server/src/lib.rs

```diff
@@ -22,30 +24,85 @@ pub mod prover {
     tonic::include_proto!("prover");
 }
 
-fn run_cairo_program_in_proof_mode(
-    execution_request: &ExecutionRequest,
-) -> Result<ExecutionResponse, Status> {
-    let (cairo_runner, vm) = run_in_proof_mode(&execution_request.program)
-        .map_err(|e| Status::internal(format!("Failed to run Cairo program: {e}")))?;
-    extract_run_artifacts(cairo_runner, vm).map_err(|e| Status::internal(e.to_string()))
+fn run_cairo_program_in_proof_mode(program: &[u8]) -> Result<ExecutionArtifacts, ExecutionError> {
+    let (cairo_runner, vm) = run_in_proof_mode(program)?;
+    extract_execution_artifacts(cairo_runner, vm)
 }
 
-async fn call_prover(prover_request: &ProverRequest) -> Result<Proof, ProverError> {
-    let public_input: PublicInput = serde_json::from_str(&prover_request.public_input)?;
-    let prover_config: ProverConfig = serde_json::from_str(&prover_request.prover_config)?;
-    let prover_parameters: ProverParameters =
-        serde_json::from_str(&prover_request.prover_parameters)?;
-
+async fn call_prover(
+    execution_artifacts: &ExecutionArtifacts,
+    prover_config: &ProverConfig,
+    prover_parameters: &ProverParameters,
+) -> Result<Proof, ProverError> {
     run_prover_async(
-        &public_input,
-        &prover_request.memory,
-        &prover_request.trace,
-        &prover_config,
-        &prover_parameters,
+        &execution_artifacts.public_input,
+        &execution_artifacts.memory,
+        &execution_artifacts.trace,
+        prover_config,
+        prover_parameters,
     )
     .await
 }
 
+fn execution_error_to_status(e: ExecutionError) -> Status {
```

Comment:
Would an `Into` or `TryInto` impl be appropriate here?

Reply:
Apparently I can even `impl From<ExecutionError> for Status`. Fixed in the latest commit.